### ISSUE TITLE
Select specific song from array

### ISSUE THREAD

**Issue by csantiago132** [benwiley4000/cassette#314]
Hi and great job on this package! It’s literally “no-config” I’ve read the docs and looked all over the repo to find a way to play a specific song from a given array. 

Use-case: I display 20 songs to the user and the user wants to play the 5th song out of the 20 it can select. 

Right now, I provide the array of songs I can’t find a way to make it play from the nth-song and forward.

**Comment by csantiago132** [comment]
If there’s no a way, and you can give me some guidance, I’ll open up a PR for it

**Comment by benwiley4000** [comment]
> It's literally "no-config"
I understand you meant this as a joke but ivet spent literally hundreds of hours managing different config options so I can assure you there is config, just maybe not the config you were looking for. 😄 Although in this case I think you might be covered!

Are you using the alpha? If so this is pretty simple to do using the `playerContext`. If you look at the docs for playerContext (https://github.com/benwiley4000/cassette/blob/next/packages/core/docs/playerContext.md) you can do:

```js
onSelectTrackIndex(playlist.findIndex(track => track.url === mySelectedUrl))
```

I might have overlooked a use case for which this doesn't work well. But in general I figured it's better to expose an API for selecting the index since there might be a number of different criteria you would need for selecting a particular track index. What do you think?

**Comment by benwiley4000** [comment]
Also I think this could be better documented. If you have any suggestions for that or want to contribute docs that is always welcome! And as I just mentioned, the API might require tweaking if you think I've overlooked something.

**Comment by csantiago132** [comment]
Hi and I mean no offense 😅😅 by that comment haha, no I’m not on alpha, I installed v1 actually because typescript gives me an error about the v2 not being called in a constructor or something like that and I literally passed the array and hit “play” and bam, the audio was playing. More than happy to contribute once I wrap my head around it. Let me take a look at the link you gave me and see how I can turn off that warning on typescript

**Comment by benwiley4000** [comment]
@csantiago132 would you mind sharing a screenshot of the error log for v2 and maybe show me the line in the library that triggered the error?

As far as I can remember there's unfortunately not a great way to do what you want in v1. The only way I can think is to set the playlist to a single track and dynamically change the playlist prop to the track you want.

**Comment by benwiley4000** [comment]
The issue came up before and that's what the user who had the issue did, I think. https://github.com/benwiley4000/cassette/issues/65

**Comment by csantiago132** [comment]
@benwiley4000 yep thats what im doing right now, im digging aorund the code to see where i can start adding types for this package (i really like it) so typescript stop giving me errors

**Comment by benwiley4000** [comment]
Cool! Again, if you think it's something I can help solve, I would love to see any error messages. :) Thanks for trying this out!

**Comment by csantiago132** [comment]
with v1, I have to import to react as 
```tsx
const AudioPlayer = require('react-responsive-audio-player');
```

doing an ES6 import triggers a warning due to typings not being present (I want to add them and propose a PR for it, im currently working on my capstone project for my bootcamp) 
```tsx
[ts]
Could not find a declaration file for module 'react-responsive-audio-player'. '/node_modules/react-responsive-audio-player/dist/audioplayer.js' implicitly has an 'any' type.

  Try `npm install @types/react-responsive-audio-player` if it exists or 
add a new declaration (.d.ts) file containing 
`declare module 'react-responsive-audio-player';` [7016]
```

declaring a module doesnt work as itll give you this error 
```tsx
[ts] Invalid module name in augmentation. Module '  this one ' resolves to an untyped module at '{project path}\node_modules\index.js', which cannot be augmented.
```

**Comment by benwiley4000** [comment]
Not sure how to explain the es6 import for v1 not working in typescript. I know it works in Babel, but it's probably not spec-compliant.

**Comment by csantiago132** [comment]
>Sorry, cassette is a different library that I don't own... you need to:


sorry im a little confused, the repo says `benwiley4000/cassette` ok ill do this and keep trying

**Comment by benwiley4000** [comment]
benwiley4000/cassette is the name on github but it's published to npm as three different packages under the `@cassette` namespace. `@cassette/core`, `@cassette/components` and `@cassette/player`. Sorry this wasn't clear... I'll try to update the documentation more soon! And if you feel like contributing to the docs I will always be eager to take PRs. :)

**Comment by benwiley4000** [comment]
@csantiago132 yep! here is a working example code sandbox... hopefully this helps you out! https://codesandbox.io/s/0y3l4pnn7n

**Comment by benwiley4000** [comment]
@csantiago132 

Note that in this example the `MediaPlayer` is split apart into `PlayerContextProvider` and `MediaPlayerControls` which means you get some more specific control over the UI. `MediaPlayer` works out of the box but without the type of control I think you're looking for.

However it depends on what type of UI you're trying to implement. If you want to add a custom button to the default controls bar, then you should pass a render prop to `MediaPlayer` and you'll have access to the playerContext. For example:

```jsx
<MediaPlayer
  playlist={playlist}
  controls={[
    'spacer',
    'backskip',
    'playpause',
    'forwardskip',
    'spacer',
    'progress',
    (playerContext) =>
      <MyTrackSelector playlist={playlist} selectTrackIndex={onSelectTrackIndex} />
  ]}
/>
```

**Comment by benwiley4000** [comment]
Again if there's anything you think could be more clear in the docs, please let me know! :)

**Comment by csantiago132** [comment]
@benwiley4000 wow, both examples are great! I implemented the first one already, im just passing 1 song to the array just to avoid rewriting too much at this point. thank you again, seeing these examples greatly help! have you heard of Docz? its pretty good for documenting as it imports code that you can use as example without too much hassle

**Comment by benwiley4000** [comment]
I haven't heard of Docz. I'm currently using react-styleguidist for the docs website. It works pretty well because I can write jsx examples as markdown files and it turns them into running interactive demos in the docs.

I'll probably stick with that for now since I've invested a lot into the site config but if it's not working for me I might check out Docz.. thanks for the suggestion!

**Comment by csantiago132** [comment]
sure anytime!
